cvat-core/src/object-utils.ts

```diff
@@ -356,3 +358,60 @@ export function rle2Mask(rle: number[], width: number, height: number): number[]
 
     return decoded;
 }
+
+export function propagateShapes<T extends SerializedShape | ObjectState>(
+    shapes: T[], from: number, to: number,
+): T[] {
+    const getCopy = (shape: T): any => {
```

Comment:
Probably return type is `SerializedData` from `cvat-core/src/object-state.ts`

Reply:
Well, its actually `SerializedShape | SerializedData` depending on the input datatype